src/embedding.rs

```diff
@@ -191,6 +277,88 @@ impl MatrixWrapper for MMapMatrix {
         }
     }
 
+    fn init_with_evec(
+        rows: usize,
+        cols: usize,
+        sparse_matrix_reader: Arc<SparseMatrix>,
+        order: TruncatedOrder
+    ) -> Self {
+        let uuid = Uuid::new_v4();
+        let file_name = format!("{}_matrix_{}", sparse_matrix_reader.get_id(), uuid);
+        let mut mmap = create_mmap(rows, cols, file_name.as_str());
+
+        let matrix_transform = |x: ArrayView2<f32>| -> Array2<f32> {
+            let shape = x.shape();
+            let mut arr = Array2::zeros((shape[0], shape[1])); 
+    
+            let sparse_laplacian_entries: Vec<Entry> = sparse_matrix_reader.entries.clone() // Might be a problem?
+            .into_par_iter()
+            .map(|entry| {
+                if entry.col == entry.row { // What happens for reflexive columns?
+                    Entry {
+                        row: entry.row,
+                        col: entry.col,
+                        value: sparse_matrix_reader.get_row_sum(entry.row),
+                    }
+                } else {
+                    Entry {
+                        row: entry.row,
+                        col: entry.col,
+                        value: - entry.value,
+                    }
+                }
+            })
+            .collect();
+
+            for entry in sparse_laplacian_entries {
+                for i in 0..shape[0] {
+                    arr[[entry.row as usize, i as usize]] += entry.value * x[[entry.col as usize, i as usize]];
+                }
+            }
+        
+            arr
+        };
+
+        let mut offset_identity_matrix: Array2<f32> = Array2::zeros((rows, cols));
+        for i in 0..std::cmp::min(rows, cols) {
+            offset_identity_matrix[[i, i]] = 1f32;
+        }
+
+        match lobpcg(
+            matrix_transform,
+            offset_identity_matrix,
+            |_x: ArrayViewMut2<f32>| {},
+            Option::None,
+            0.1f32,
```

Comment:
Extract magical values to variables (so we know what they represent reading the code without needing to jump into lobpcg function definition.